packages/flutter/lib/src/material/pickers/date_range_picker_dialog.dart

```diff
@@ -0,0 +1,678 @@
+          onEndDateChanged: _handleEndDateChanged,
+          onConfirm: _hasSelectedDateRange ? _handleOk : null,
+          onCancel: _handleCancel,
+          onToggleEntryMode: _handleEntryModeToggle,
+          // TODO(darrenaustin): localize 'SAVE'
+          confirmText: widget.saveText ?? 'SAVE',
+          // TODO(darrenaustin): localize 'SELECTED RANGE'
+          helpText: widget.helpText ?? 'SELECTED RANGE',
+        );
+        size = mediaQuery.size;
+        insetPadding = const EdgeInsets.all(0.0);
+        shape = const RoundedRectangleBorder(
+          borderRadius: BorderRadius.all(Radius.circular(0.0))
+        );
+        break;
+
+      case DatePickerEntryMode.input:
+        contents = _InputDateRangePickerDialog(
+          selectedStartDate: _selectedStart,
+          selectedEndDate: _selectedEnd,
+          picker: Container(
+            padding: const EdgeInsets.symmetric(horizontal: 24),
+            height: orientation == Orientation.portrait
+              ? _inputFormPortraitHeight
+              : _inputFormLandscapeHeight,
+            child: Column(
+              children: <Widget>[
+                const Spacer(),
+                InputDateRangePicker(
+                  key: _inputPickerKey,
+                  initialStartDate: _selectedStart,
+                  initialEndDate: _selectedEnd,
+                  firstDate: widget.firstDate,
+                  lastDate: widget.lastDate,
+                  onStartDateChanged: _handleStartDateChanged,
+                  onEndDateChanged: _handleEndDateChanged,
+                  autofocus: true,
+                  autovalidate: _autoValidate,
+                  helpText: widget.helpText,
+                  errorInvalidRangeText: widget.errorInvalidRangeText,
+                  errorFormatText: widget.errorFormatText,
+                  errorInvalidText: widget.errorInvalidText,
+                  fieldStartHintText: widget.fieldStartHintText,
+                  fieldEndHintText: widget.fieldEndHintText,
+                  fieldStartLabelText: widget.fieldStartLabelText,
+                  fieldEndLabelText: widget.fieldEndLabelText,
+                ),
+                const Spacer(),
+              ],
+            ),
+          ),
+          onConfirm: _handleOk,
+          onCancel: _handleCancel,
+          onToggleEntryMode: _handleEntryModeToggle,
+          confirmText: widget.confirmText ?? localizations.okButtonLabel,
+          cancelText: widget.cancelText ?? localizations.cancelButtonLabel,
+          // TODO(darrenaustin): localize 'SELECTED DATE RANGE'
+          helpText: widget.helpText ?? 'SELECTED DATE RANGE',
+        );
+        size = orientation == Orientation.portrait ? _inputPortraitDialogSize : _inputLandscapeDialogSize;
+        insetPadding = const EdgeInsets.symmetric(horizontal: 16.0, vertical: 24.0);
+        shape = const RoundedRectangleBorder(
+          borderRadius: BorderRadius.all(Radius.circular(4.0))
```

Comment:
There will be theming support coming in the future right? We'd want to be able to customize the corner radius for example.

Reply:
Yup, I updated this to defer to the shape from the theme if there is one.